axelor-account/src/main/java/com/axelor/apps/account/service/ReconcileGroupServiceImpl.java

```diff
@@ -77,12 +77,49 @@ public void validate(ReconcileGroup reconcileGroup, List<Reconcile> reconcileLis
           reconcileGroup);
     }
 
-    reconcileGroup.setStatusSelect(ReconcileGroupRepository.STATUS_FINAL);
+    reconcileGroup.setStatusSelect(ReconcileGroupRepository.STATUS_BALANCED);
     reconcileGroup.setDateOfLettering(appBaseService.getTodayDate(reconcileGroup.getCompany()));
 
     reconcileGroupSequenceService.fillCodeFromSequence(reconcileGroup);
   }
 
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void cancelProposal(ReconcileGroup reconcileGroup) {
+    if (reconcileGroup != null) {
+      if (reconcileGroup.getStatusSelect() == ReconcileGroupRepository.STATUS_PROPOSAL) {
+        remove(reconcileGroup);
+      } else if (reconcileGroup.getStatusSelect() == ReconcileGroupRepository.STATUS_PARTIAL) {
+        List<Reconcile> reconcileList =
+            reconcileRepository
+                .all()
+                .filter(
+                    "self.reconcileGroup.id = :reconcileGroupId AND self.statusSelect = :statusDraft")
+                .bind("reconcileGroupId", reconcileGroup.getId())
+                .bind("statusDraft", ReconcileRepository.STATUS_DRAFT)
+                .fetch();
+        for (Reconcile reconcile : reconcileList) {
+          reconcileRepository.remove(reconcile);
+        }
```

Comment:
You also need to remove the ReconcileGroup on the MoveLines related to the Reconciles you delete and set the ReconcileGroup.isProposal to false